### PR TITLE
api: add comments on fields for reference docs

### PR DESCRIPTION
**Pull request by mtardy**
Fixes https://github.com/cilium/tetragon/issues/957.

This PR includes no functional changes. Only comments for autogenerated documentation.
I need your help with a few fields.

### CODE REVIEW

File: api/v1/tetragon/tetragon.proto [hunk 1]
```diff
@@ -45,71 +50,128 @@ message Container {
...
     string flags = 7;
+    // Start time of the execution.
     google.protobuf.Timestamp start_time = 8;
+    // TODO: what is this identifier about? related to uid apparently.
     google.protobuf.UInt32Value auid = 9;
+    // Information about the the Kubernetes Pod where the event originated.
     Pod pod = 10;
+    // TODO: is this the 15 first digits of the docker ID? We should name it
+    // something else than docker since most k8s install no longer use Docker.
     string docker = 11;
```
Comment (mtardy) [inline api/v1/tetragon/tetragon.proto:139]: Need minor help here!

File: api/v1/tetragon/tetragon.proto [hunk 2]
```diff
@@ -45,71 +50,128 @@ message Container {
...
+    // Exec ID of the parent process.
     string parent_exec_id = 12;
+    // TODO: is this a similar refcnt than from the BPF stats?
     uint32 refcnt = 13;
+    // Set of capabilities that define the permissions the process can execute with.
     Capabilities cap = 14;
+    // TODO: is this used?
     Namespaces ns = 15;
```
Comment (mtardy) [inline api/v1/tetragon/tetragon.proto:147]: Do you know when it's used?

File: api/v1/tetragon/tetragon.proto [hunk 3]
```diff
@@ -45,71 +50,128 @@ message Container {
...
     string cwd = 4;
+    // Absolute path of the executed binary.
     string binary = 5;
+    // Arguments passed to the binary at execution.
     string arguments = 6;
+    // Information about which syscalls generated events, use of internal
+    // Tetragon buffers, errors and more.
+    // TODO: not sure how to documents all the existing flags without dumping
+    // 60 lines of text
     string flags = 7;
```
Comment (mtardy) [inline api/v1/tetragon/tetragon.proto:130]: What do you think?

File: api/v1/tetragon/tetragon.proto [hunk 4]
```diff
@@ -45,71 +50,128 @@ message Container {
...
+    // Start time of the execution.
     google.protobuf.Timestamp start_time = 8;
+    // TODO: what is this identifier about? related to uid apparently.
     google.protobuf.UInt32Value auid = 9;
```
[the remark this answers is not in the document]
Comment (willfindlay) [reply api/v1/tetragon/tetragon.proto:134]: auid is audit UID

File: api/v1/tetragon/tetragon.proto (hunk 1 above)
Comment (willfindlay) [reply api/v1/tetragon/tetragon.proto:139]: Yep this is correct. And yes it should probably be deprecated and renamed.

File: api/v1/tetragon/tetragon.proto [hunk 5]
```diff
@@ -45,71 +50,128 @@ message Container {
...
+    // Information about the the Kubernetes Pod where the event originated.
     Pod pod = 10;
+    // TODO: is this the 15 first digits of the docker ID? We should name it
+    // something else than docker since most k8s install no longer use Docker.
     string docker = 11;
+    // Exec ID of the parent process.
     string parent_exec_id = 12;
+    // TODO: is this a similar refcnt than from the BPF stats?
     uint32 refcnt = 13;
```
[the remark this answers is not in the document]
Comment (willfindlay) [reply api/v1/tetragon/tetragon.proto:143]: I believe this is the processCache reference count. When it hits zero, the entry is evicted from the processCache. https://github.com/cilium/tetragon/blob/main/pkg/process/cache.go#L37
https://github.com/cilium/tetragon/blob/bc5f11a52c6bc19607dc6c15aa7fe44894e86a1b/pkg/process/process.go#L91

File: api/v1/tetragon/tetragon.proto (hunk 2 above)
Comment (willfindlay) [reply api/v1/tetragon/tetragon.proto:147]: https://github.com/cilium/tetragon/blob/bc5f11a52c6bc19607dc6c15aa7fe44894e86a1b/pkg/process/process.go#L134-L136

File: api/v1/tetragon/tetragon.proto (hunk 3 above)
Comment (willfindlay) [reply api/v1/tetragon/tetragon.proto:130]: It's too bad this is a string instead of a protobuf enum :/ probably too late to change it sadly.

**Comment by mtardy** [comment]
Thanks @willfindlay, I added some of your API changes suggestion there https://github.com/cilium/tetragon/issues/486#issuecomment-1431019714.

File: api/v1/tetragon/tetragon.proto (hunk 2 above)
Comment (mtardy) [reply api/v1/tetragon/tetragon.proto:147]: ah okay there is a `option.Config.EnableProcessNs`

**Review by sharlns — APPROVED** [review]
Amazing, LGTM. Thank you! :)

File: api/v1/tetragon/tetragon.proto [hunk 6]
```diff
@@ -45,71 +50,178 @@ message Container {
...
+    // - `rootCWD` Indicates that CWD is the root directory. This is necessary
+    // to inform readers the CWD is not in the event buffer and is '/' instead.
+    // - `errorCWD` An error flag indicating an error occurred while reading
+    // the CWD of a process. If this is set it should be reported to Tetragon
+    // developers for debugging.
+    // - `clone` Indicates the process issued a clone before exec*. This is the
+    // general flow to exec* a new process, however its possible to replace the
+    // current process with a new process by doing an exec* without a clone. In
+    // this case the flag will be omitted and the same PID will be used by the
+    // kernel for both the old process and the newly exec'd process.
```
Comment (tixxdz) [inline api/v1/tetragon/tetragon.proto:177]: This field is somehow for debugging, it contains some values that may not change but others may...

Would be better to add something like:
The flags is for debugging purposes only and should not be relied upon to propagate comprehensive information.

Comment (mtardy) [reply api/v1/tetragon/tetragon.proto:177]: when you say "add" you mean append this at the end or at the beginning or replacing the whole flag documentation with this?

Comment (tixxdz) [reply api/v1/tetragon/tetragon.proto:177]: I'm fine adding it at the beginning so it is clear and we keep current doc that you added, the developers are on their own here... ;-)

Comment (mtardy) [reply api/v1/tetragon/tetragon.proto:179]: @tixxdz see https://github.com/cilium/tetragon/compare/f4a972a98319ec1f9608f835d04d8524b0aefeda..46156cca63c72f36073b79d8deddfee603b66d69.

**Comment by mtardy** [comment]
Check failure is related to dependabot.yaml config format fails and I have an open PR to fix that. Let's merge this.

File: api/v1/tetragon/tetragon.proto (hunk 3 above)
Comment (tixxdz) [reply api/v1/tetragon/tetragon.proto]: @willfindlay we just made a more of debugging flag, we can add a more reliable field later and get rid of this one if developers or other tools request this... so I guess we are fine ;-) !

**Pull request merged**